src/Illuminate/Support/Siesta.php

```diff
@@ -0,0 +1,383 @@
+    }
+
+    /**
+     * Sleep for the duration in seconds.
+     *
+     * @param  int|float  $duration
+     * @return static
+     */
+    public static function sleep($duration)
+    {
+        return (new static($duration))->seconds();
+    }
+
+    /**
+     * Sleep for the duration in minutes.
+     *
+     * @return $this
+     */
+    public function minutes()
+    {
+        $this->duration->add('minutes', $this->pullPending());
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in minutes.
+     *
+     * @return $this
+     */
+    public function minute()
+    {
+        return $this->minutes();
+    }
+
+    /**
+     * Sleep for the duration in seconds.
+     *
+     * @return $this
+     */
+    public function seconds()
+    {
+        $this->duration->add('seconds', $this->pullPending());
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in seconds.
+     *
+     * @return $this
+     */
+    public function second()
+    {
+        return $this->seconds();
+    }
+
+    /**
+     * Sleep for the duration in milliseconds.
+     *
+     * @return $this
+     */
+    public function milliseconds()
+    {
+        $this->duration->add('milliseconds', $this->pullPending());
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in milliseconds.
+     *
+     * @return $this
+     */
+    public function millisecond()
+    {
+        return $this->milliseconds();
+    }
+
+    /**
+     * Sleep for the duration in microseconds.
+     *
+     * @return $this
+     */
+    public function microseconds()
+    {
+        $this->duration->add('microseconds', $this->pullPending());
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in microseconds.
+     *
+     * @return $this
+     */
+    public function microsecond()
+    {
+        return $this->microseconds();
+    }
+
+    /**
+     * Add additional time to sleep for.
+     *
+     * @param  int|float  $duration
+     * @return $this
+     */
+    public function and($duration)
+    {
+        $this->pending = $duration;
+
+        return $this;
+    }
+
+    /**
+     * Handle the object's destruction.
+     *
+     * @return void
+     */
+    public function __destruct()
+    {
+        if (! $this->shouldSleep) {
+            return;
+        }
+
+        if ($this->pending !== null) {
+            throw new RuntimeException('Unknown duration unit.');
+        }
+
+        if (static::$fake) {
+            static::$sequence[] = $this->duration;
+
+            return;
+        }
+
+        $remaining = $this->duration->copy();
+
+        $seconds = (int) $remaining->totalSeconds;
+
+        if ($seconds > 0) {
+            sleep($seconds);
+
+            $remaining = $remaining->subSeconds($seconds);
+        }
+
+        $microseconds = (int) $remaining->totalMicroseconds;
+
+        if ($microseconds > 0) {
+            usleep($microseconds);
+        }
```

Comment:
This can be simplified imo. In one of my projects I have added a helper that combines `sleep()` and `usleep()` so i can do fractional sleep easily like so:
```php
function my_sleep($microseconds)
    {
        if ($microseconds <= 0) {
            return;
        }
        if ($usleep = $microseconds % 1000000) {
            usleep($usleep);
        }

        if ($sleep = ($microseconds - $usleep) / 1000000) {
            sleep($sleep);
        }
    }
```

Then you can call that method (maybe even expose it as public/static since it is very handy) and be done with it. The `CarbonInterval $duration` property in your class is probably too complex since you can just use a microseconds integer instead. We are mostly talking about times in that neighborhood anyway (maybe up to a few seconds).